templates/pritunl-plugin-ldap.j2

```diff
@@ -0,0 +1,118 @@
+import ldap
+import os
+import pritunlsdk
+from pritunl import logger
+
+# LDAPSearch README
+# https://github.com/chaordic/python-ldap-simple-search
+
+class LDAPSearch(object):
+    _conn = None
+
+    def __init__(self,
+                 uri,
+                 bind_dn,
+                 password):
+
+        ldap.set_option(ldap.OPT_REFERRALS, 0)
+        self.uri = uri
+        self.bind_dn = bind_dn
+        self.password = password
+
+    def __del__(self):
+        self._unbind()
+
+    def conn(self):
+        if self._conn is None:
+            self._conn = self._bind()
+        return self._conn
+
+    def _bind(self):
+        l = ldap.initialize(self.uri)
+        l.protocol_version = ldap.VERSION3
+        try:
+            l.simple_bind_s(self.bind_dn, self.password)
+        except ldap.INVALID_CREDENTIALS:
+            logger.error('Your username or password is incorrect.')
+            raise
+        except ldap.LDAPError as e:
+            if type(e.message) == dict and 'desc' in e.message:
+                logger.error(e.message['desc'])
+            if type(e.message) == dict and 'desc' in e.message:
+                logger.error(e)
+            else:
+                logger.error(e)
+            raise
+        return l
+
+    def _unbind(self):
+        if self._conn is not None:
+            self.conn().unbind_s()
```

Comment:
What about to call the class attr `self._conn` directly? Eg.: `self._conn.unbind_s()` 

In general in OOP, we use methods to access resources outside the class.

Reply:
You mean call ```self._conn.unbind_s()``` instead ```_unbind()```? I'm not using it outside the class. I call this internal method because before to unbind ldap I need to check the connection with it.